pkg/operator/reports.go

```diff
@@ -859,6 +934,57 @@ func (op *Reporting) queueDependentReportsForReport(report *metering.Report) err
 	return nil
 }
 
+func (op *Reporting) doReportsDependOnThisReport(report *metering.Report) bool {
+	// Look for all reports in the namespace
+	reports, err := op.reportLister.Reports(report.Namespace).List(labels.Everything())
+	if err != nil {
+		op.logger.Errorf("was unable to list reports for report: %s, while looking for dependent reports", report.Name)
+		return false
+	}
+
+	// for each report in the namespace, find ones that depend on the report passed into the function.
+	for _, otherReport := range reports {
+		deps, err := op.getReportDependencies(otherReport)
+		if err != nil {
+			op.logger.Errorf("was unable to get dependencies for report: %s, while looking for dependent reports", report.Name)
+			return false
+		}
+		// If a Report has a dependency on the passed in report, we're done
+		for _, dep := range deps.Reports {
+			if dep.Name == report.Name {
+				break
+			}
+		}
```

Comment:
Just trying to get some clarity in the situation where we're creating something like a monthly roll-up Report, which depends on a weekly scheduled Report as an input to that object, and the weekly scheduled Report has a configured retention period: if I'm reading this correctly, we essentially don't mark that object as "expired" as there's another Report (the monthly scheduled one) marked as a dependent. What happens in the case where the outer report doesn't have a retention period and the user hasn't explicitly deleted this object? It seems like the inner report will just keep getting marked as un-expired and re-processed in the queue.

Reply:
And I'm sure the immediate answer could be to make the documentation explicit enough to ensure that both the inner and outer report have a configured retention period but just wanted to see if I was reading that situation correctly.

Reply:
Maybe this could be a validation check too, e.g. if the outer report object depends on a report input, ensure that the outer report has a configured retention period if the report dependency does also?

Reply:
Not sure about the inner/outer terminology, but yes, it's easy to get in a situation where a Report won't be going away because something depends on it. We should document this.

On validation, probably. A Report added with an expiration period set which is depended on by a Report without one: I can't imagine a situation where this would be desired.

The only downside is the validation requires scanning the dependency tree of the Reports. From that scan it may be hard to clearly tell the user why the validation is failing, for instance, if it's failing validation because of two or more dependent Reports. But I suppose just knowing about the problem with any feedback is better than just wondering why the Report never goes away?

Reply:
Yeah agreed, it's annoying now knowing the totality of the situation, but we should aim towards at least providing some debugging mechanisms like container logs and events that can help portray the state these objects are in, why they aren't being deleted despite reaching the expiration date, etc.

Reply:
yep, so towards that end, added logging on the thing depended on when we fail to delete report for dependencies. Done in bf132bc2